### ISSUE TITLE
Transaction errors should emit events to the sidecar

### ISSUE THREAD

**Issue by kantai** [blockstack/stacks-blockchain#1511]
Transaction errors that occur after a transaction has been included in the mempool, but result in the transaction _not_ being valid for block inclusion should emit events to the sidecar.

This class of errors _should_ be minimal, however, we should be able to alert the sidecar to them in the event that they do occur.

@zone117x

**Comment by jcnelson** [comment]
> but result in the transaction not being valid for block inclusion should emit events to the sidecar.

I'm confused by this statement.  Are we talking about transactions that get rendered unmineable by the inclusion of some conflicting transaction?  If so, then I can see the node emitting an error for transactions that get orphaned (e.g. a transaction with an equal nonce gets mined by the same account), but I can't see how this would work for the general case (e.g. a transaction gets orphaned because its post-conditions can no longer be true).

Only miners have insight into which transactions can't be mined in the general case.  Other nodes only see the result of miners' local, private decisions when they process the blocks.

**Comment by zone117x** [comment]
I'm seeing this ghost transaction behavior more often. Would really like for there to be a way to see what is going wrong without having to dig into sqlite or huge debug logs

**Comment by zone117x** [comment]
Attached are debug logs from submitting a tx several time (like 5?) and receiving a 200 w/ tx_id from the core_node, but nothing happening. Account nonce was also not incrementing. 

[wtf.txt](https://github.com/blockstack/stacks-blockchain/files/4560332/wtf.txt)

**Comment by jcnelson** [comment]
That would be consistent with the transaction not being mined.  Notice the lines that look like this:

```
stacks-blockchain_1  | 2020-04-30T18:06:08.208549202Z DEBUG [1588269968.208] [src/chainstate/stacks/db/blocks.rs:2689] Process block "fe944712ab3d0fd343e2e4da0e5fdb9a68dc951727a9b0dc512cb5868b58753b" with 1 transactions
```

It's always `with 1 transactions`.

That you get a txid back only tells you the tx made it into the mempool at some point in the last 200 blocks.  It doesn't tell you whether or not it will ever be mined (but this is true for all blockchains, including ours).  Also, if you submit the same transaction repeatedly, the node will just give you back the txid and take no further action.

I'm not seeing any log messages that would indicate that the node is attempting to relay the transaction, which tells me that the transaction was already in the mempool for the duration of time covered by this log (the node will only relay new transactions it sees).

**Comment by kantai** [comment]
I believe the issue in these logs is that this is running on `mocknet`, and the mocknet driver isn't polling the mempool correctly. Opening a PR now.

**Comment by kantai** [comment]
Since the issue that prompted this was resolved in #1517, I'm going to close this.

I think if we want a system for detecting "low fee" transactions that are hanging around in the mempool without getting confirmed, that'll likely be a separate system anyways.
